Index: linux-3.10.108/drivers/net/wireless/mtk/mt7603/mt7603_wifi/Kconfig
===================================================================
--- linux-3.10.108.orig/drivers/net/wireless/mtk/mt7603/mt7603_wifi/Kconfig	2018-09-20 18:41:43.501329000 +0800
+++ linux-3.10.108/drivers/net/wireless/mtk/mt7603/mt7603_wifi/Kconfig	2018-09-20 18:44:29.318568000 +0800
@@ -410,7 +410,7 @@
     endchoice
 
     if MT7603E_WIFI_MODE_AP || MT7603E_WIFI_MODE_BOTH
-        source "drivers/net/wireless/mtk/mt7603e/mt7603_wifi_ap/Kconfig"
+        source "drivers/net/wireless/mtk/mt7603/mt7603_wifi_ap/Kconfig"
     endif
 
     #if MT7603E_WIFI_MODE_STA || MT7603E_WIFI_MODE_BOTH
Index: linux-3.10.108/drivers/net/wireless/mtk/mt7603/mt7603_wifi/os/linux/rt_profile.c
===================================================================
--- linux-3.10.108.orig/drivers/net/wireless/mtk/mt7603/mt7603_wifi/os/linux/rt_profile.c	2018-09-20 18:41:48.142313000 +0800
+++ linux-3.10.108/drivers/net/wireless/mtk/mt7603/mt7603_wifi/os/linux/rt_profile.c	2018-09-20 18:45:03.392296000 +0800
@@ -54,8 +54,8 @@
 #define def_to_str(s)    #s
 
 #if defined(CONFIG_SUPPORT_OPENWRT)
-#define FIRST_EEPROM_FILE_PATH	"/etc/wireless/mt7603e/mt7603.eeprom.dat"
-#define FIRST_AP_PROFILE_PATH		"/etc/wireless/mt7603e/mt7603.dat"
+#define FIRST_EEPROM_FILE_PATH	"/etc/wireless/mt7603/mt7603.eeprom.dat"
+#define FIRST_AP_PROFILE_PATH		"/etc/wireless/mt7603/mt7603.dat"
 #else
 #define FIRST_EEPROM_FILE_PATH	"/etc_ro/Wireless/RT2860/"
 #define FIRST_AP_PROFILE_PATH		"/etc/Wireless/RT2860/RT2860.dat"
@@ -63,8 +63,8 @@
 #define FIRST_CHIP_ID	xdef_to_str(CONFIG_RT_FIRST_CARD)
 
 #if defined(CONFIG_SUPPORT_OPENWRT)
-#define SECOND_EEPROM_FILE_PATH	"/etc/wireless/mt7603e/mt7603.eeprom.dat"
-#define SECOND_AP_PROFILE_PATH		"/etc/wireless/mt7603e/mt7603.dat"
+#define SECOND_EEPROM_FILE_PATH	"/etc/wireless/mt7603/mt7603.eeprom.dat"
+#define SECOND_AP_PROFILE_PATH		"/etc/wireless/mt7603/mt7603.dat"
 #else
 #define SECOND_EEPROM_FILE_PATH	"/etc_ro/Wireless/iNIC/"
 #define SECOND_AP_PROFILE_PATH	"/etc/Wireless/iNIC/iNIC_ap.dat"
Index: linux-3.10.108/drivers/net/wireless/mtk/mt7603/mt7603_wifi_ap/Makefile
===================================================================
--- linux-3.10.108.orig/drivers/net/wireless/mtk/mt7603/mt7603_wifi_ap/Makefile	2018-09-20 18:41:49.003274000 +0800
+++ linux-3.10.108/drivers/net/wireless/mtk/mt7603/mt7603_wifi_ap/Makefile	2018-09-20 18:44:19.682781000 +0800
@@ -1,7 +1,7 @@
 ifeq ($(CONFIG_SUPPORT_OPENWRT),y)
 ccflags-y = -I$(src)/../mt7603_wifi/include \
 		-I$(src)/../mt7603_wifi/ate/include
-DRV_NAME = mt7603e
+DRV_NAME = mt7603
 RT_WIFI_DIR = ../mt7603_wifi
 
 else
